controllers/managedocs_controller.go

```diff
@@ -1297,11 +1297,19 @@ func (r *ManagedOCSReconciler) reconcileRookCephOperatorConfig() error {
 		rookConfigMap.Data["CSI_RBD_PLUGIN_RESOURCE"] = rbdPluginRequirements
 		rookConfigMap.Data["CSI_CEPHFS_PROVISIONER_RESOURCE"] = fsProvisionerRequirements
 		rookConfigMap.Data["CSI_CEPHFS_PLUGIN_RESOURCE"] = fsPluginRequirements
+	}
 
-		if err := r.update(rookConfigMap); err != nil {
-			return fmt.Errorf("Failed to update Rook ConfigMap: %v", err)
+	// Update rookConfigMap only for provider deployment
+	// Disable ceph-csi on provider deployment
+	if r.DeploymentType == providerDeploymentType {
+		if rookConfigMap.Data["ROOK_CSI_ENABLE_CEPHFS"] != "false" ||
+			rookConfigMap.Data["ROOK_CSI_ENABLE_RBD"] != "false" {
```

Comment:
why this check is required? you can directly assign "false" value if it's a providerDeploymentType?